ownCloud/Client/Actions/Actions+Extensions/CollaborateAction.swift

```diff
@@ -24,13 +24,29 @@ class CollaborateAction: Action {
 	override class var identifier : OCExtensionIdentifier? { return OCExtensionIdentifier("com.owncloud.action.collaborate") }
 	override class var category : ActionCategory? { return .normal }
 	override class var name : String { return OCLocalizedString("Sharing", nil) }
-	override class var locations : [OCExtensionLocationIdentifier]? { return [.keyboardShortcut, .contextMenuSharingItem, .moreItem, .moreDetailItem, .accessibilityCustomAction] }
+	override class var locations : [OCExtensionLocationIdentifier]? { return [.keyboardShortcut, .contextMenuSharingItem, .moreItem, .moreFolder, .moreDetailItem, .accessibilityCustomAction] }
 	override class var keyCommand : String? { return "S" }
 	override class var keyModifierFlags: UIKeyModifierFlags? { return [.command] }
 
 	// MARK: - Extension matching
 	override class func applicablePosition(forContext: ActionContext) -> ActionPosition {
 		if forContext.items.count == 1, let core = forContext.core, core.connectionStatus == .online, core.connection.capabilities?.sharingAPIEnabled == 1, let item = forContext.items.first, item.isShareable {
+			// For .moreFolder location: Check special cases at root level
```

Comment:
The checks carried out here should not be necessary as the cases it tries to handle should already be prevented by checking `item.isShareable`, which relies on the permissions returned by the server.

The action should also not show up for root folders even without that code. Can you verify/reproduce?